src/core/tests/Test_CoreMain.py

```diff
@@ -1143,6 +1168,624 @@ def __check_telemetry_events(self, runtime):
+        # {\"patchId\": \"libgcc_5.60.7-8.1_Ubuntu_16.04\", \"name\": \"libgcc\", \"version\": \"5.60.7-8.1\", \"classifications\": [\"Other\"]},
+        # {\"patchId\": \"libgoa-1_0-0_3.20.5-9.6_Ubuntu_16.04\", \"name\": \"libgoa-1_0-0\", \"version\": \"3.20.5-9.6\", \"classifications\": [\"Other\"]}
+
+        patch_count = 99997
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        runtime.status_handler.set_package_assessment_status(test_packages, test_package_versions, "Security")
+        runtime.status_handler.set_assessment_substatus_json(status=Constants.STATUS_SUCCESS)
+
+        # Assert complete status file
+        with runtime.env_layer.file_system.open(runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        self.assertTrue(len(json.dumps(substatus_file_data)) > Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+        self.__assert_complete_status_output(substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS, patch_count + 3)
+
+        # Assert assessment truncated status file
+        with runtime.env_layer.file_system.open(runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        self.assertTrue(len(json.dumps(substatus_file_data)) < Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+        self.__assert_truncated_status_output(substatus_file_data[0]["status"]["substatus"][0], Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_WARNING)
+        # Assert assessment tombstone
+        self.__assert_truncated_assessment_tombstone(substatus_file_data[0]["status"]["substatus"][0], patch_count + 3)
+        # Assert assessment truncated error
+        self.__assert_truncated_error(substatus_file_data[0]["status"]["substatus"][0], error_count=0)
+
+        runtime.stop()
+
+    def test_installation_truncation_over_size_limit(self):
+        argument_composer = ArgumentComposer()
+        argument_composer.operation = Constants.INSTALLATION
+        runtime = RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.ZYPPER)
+        runtime.set_legacy_test_type('SuccessInstallPath')
+        CoreMain(argument_composer.get_composed_arguments())
+
+        # Check telemetry events
+        self.__check_telemetry_events(runtime)
+
+        # SuccessInstallPath add 2 additional packages
+        # {\"patchId\": \"kernel-default_4.4.49-92.11.1_Ubuntu_16.04\", \"name\": \"kernel-default\", \"version\": \"4.4.49-92.11.1\", \"classifications\": [\"Other\"]},
+        #  {\"patchId\": \"libgoa-1_0-0_3.20.5-9.6_Ubuntu_16.04\", \"name\": \"libgoa-1_0-0\", \"version\": \"3.20.5-9.6\", \"classifications\": [\"Other\"]}
+
+        patch_count_for_assessment = random.randint(798, 1100)
```

Comment:
Wouldn't assessment patches also be truncated with this count? The test name seems to imply it's install substatus that will be truncated

Reply:
the operation is installation, it performs both. named updated to installation_operation_truncation....